book/src/e2e-testing.md

```diff
@@ -0,0 +1,81 @@
+# End to End Testing for CSI Storage Plugins
+
+Currently, csi-sanity exists to help test compliance with the spec. But e2e testing of plugins is needed as well to provide plugin authors and users stronger guarantees that their plugin "just works" in kubernetes.
+
+## Setting up End to End tests for your CSI Plugin
+
+### Prerequisites:
+ * A Kubernetes v1.12+ Cluster
+ * [Kubectl](https://kubernetes.io/docs/tasks/tools/install-kubectl/#install-kubectl)
+
+There are two ways to run end-to-end tests for your CSI Plugin
+ 1) use [Kubernetes E2E Tests](https://github.com/kubernetes/kubernetes/tree/master/test/e2e/storage/external), by providing a DriverDefinition YAML file via a parameter.
+ 2) import the in-tree storage tests and run them using `go test`. 
+ 
+ This doc will cover how to run the E2E tests using the second method. 
+ 
+### Note: 
+ 
+ In some cases you would not be able to use the first method, in running e2e tests by just providing a YAML file defining your CSI plugin.
+
+ * For example the [NFS CSI plugin](https://github.com/kubernetes-csi/csi-driver-nfs) does not support dynamic provisoning, so we would want to skip those and run only pre-provisioned tests. For such cases, you would need to write your own testdriver, which is discussed below. 
+
+In-tree storage e2e tests could be used to test CSI storage plugins. Your repo could be setup similar to how this [NFS CSI plugin](https://github.com/mathu97/csi-driver-nfs/) is setup, where the testfiles are in a test directory and the main test file is in the cmd directory. 
```

Comment:
Ya I was thinking that too. I'll work on a PR soon.

Reply:
Also the ability to import e2e tests came in Kubernetes 1.14, and it would be required for the driver to use that. The nfs driver currently uses Kubernetes 1.12, so that would need to be updated. How would this work, if I wanted to update the kubernetes version for csi-driver-nfs?

Reply:
bump everything in Gopkg.toml to *1.14 from *1.12 and dep ensure

Reply:
Created a PR to add this to the nfs driver repo. Would need to update the links.